lib/presentation/features/home/bloc/home_bloc.dart

```diff
@@ -1,47 +1,56 @@
 import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:marvelapp_flutter/domain/error_handling/exceptions.dart';
 import 'package:marvelapp_flutter/domain/use_cases/get_characters_use_case.dart';
 import 'package:marvelapp_flutter/presentation/features/home/bloc/home_event.dart';
 import 'package:marvelapp_flutter/presentation/features/home/bloc/home_state.dart';
-import 'package:marvelapp_flutter/presentation/extensions/to_character_view_data.dart';
+import 'package:marvelapp_flutter/presentation/converters/to_character_view_data.dart';
 import 'package:marvelapp_flutter/presentation/models/character_view_data.dart';
 
 class HomeBloc extends Bloc<HomeEvent, HomeState> {
   final GetCharactersUseCase getCharactersUseCase;
 
-  HomeBloc({required this.getCharactersUseCase}) : super(const HomeState(loading: false)) {
+  HomeBloc({required this.getCharactersUseCase}) : super(const HomeState(loading: Loading.firstLoading)) {
     on<ReadyForData>(_mapReadyForDataEventToState);
     on<ScrolledToEnd>(_mapScrolledToEndEventToState);
   }
 
   Future<void> _mapReadyForDataEventToState(ReadyForData event, Emitter<HomeState> emit) async {
-    emit(state.copyWith(loading: true));
+    emit(state.copyWith(loading: Loading.firstLoading));
     try {
-      final characters = await getCharactersUseCase(0);
-      final charactersViewData = characters.map((item) => item.toCharacterViewData()).toList();
-      emit(
-        state.copyWith(loading: false, characters: charactersViewData, error: null),
-      );
+      var characters = await getCharactersUseCase(0);
+      var characterViewData = characters.map((item) => item.toCharacterViewData()).toList();
+      emit(state.copyWith(loading: Loading.loaded, characters: characterViewData, error: null));
+    } on DataParsingException {
+      emit(state.copyWith(loading: Loading.loaded, error: "data parsing exception"));
+    } on DataRetrieveException {
+      emit(state.copyWith(loading: Loading.loaded, error: "data retrieve exception"));
+    } on NoConnectionException {
+      emit(state.copyWith(loading: Loading.loaded, error: "no connection exception"));
     } catch (error) {
-      emit(state.copyWith(loading: false, error: error.toString()));
+      emit(state.copyWith(loading: Loading.loaded, error: "unknown exception"));
     }
```

Comment:
There is no difference with handling these different errors, only the message that you are not using, in this case we no need all these types, only one can be enough.

Reply:
done